xdas/compose.py

```diff
@@ -0,0 +1,530 @@
+            before or after is determined by `locator`
+        atom : Atom
+            The instantiated Atom object to be inserted
+        locator : Callable
+            A function passed on from the calling Atom
+            that indicates whether the insertion is made
+            before or after `pos`
+
+        """
+
+        # List of current keys
+        keys = list(self.keys())
+        # Locate insertion position
+        pos = locator(keys, pos)
+        # Add key:item to dictionary (checks for duplicate names)
+        self.__setitem__(atom._name, atom)
+        name = atom._name
+        # Insert key at insertion position
+        keys.insert(pos, name)
+
+        # Reorder dict
+        self.data = {key: self.get(key) for key in keys}
+        pass
+
+    def _move(self, key: Hashable, locator: Callable) -> None:
+        """
+        Move an Atom within the Sequence in a direction indicated
+        by `locator`. This method is called from a parent Atom object
+        using the `Atom.move_up` or `Atom.move_down` methods.
+        Example:
+            `Sequence[key].move_down()`
+
+        Parameters
+        ----------
+        pos : Hashable
+            Key of the position at which `atom` needs
+            to be inserted. Whether the insertion is
+            before or after is determined by `locator`
+        atom : Atom
+            The instantiated Atom object to be inserted
+        locator : Callable
+            A function passed on from the calling Atom
+            that indicates whether the insertion is made
+            before or after `pos`
+
+        """
+
+        # List of current keys
+        keys = list(self.keys())
+        # Current position of key
+        current_pos = keys.index(key)
+        # Target position of key
+        target_pos = locator(keys, key)
+
+        # Moving key up
+        if current_pos == target_pos:
+            target_pos -= 1
+            current_pos += 1
+        # Moving key down
+        else:
+            target_pos += 1
+
+        # Key is already in first spot,
+        # so moving up does nothing
+        if target_pos < 0:
+            return
+
+        # Key is already in last spot,
+        # so moving down does nothing
+        if target_pos > len(keys):
+            return
+
+        # Insert key in target position
+        keys.insert(target_pos, key)
+        # Delete old key position
+        keys.pop(current_pos)
+        # Reorder dict
+        self.data = {key: self.get(key) for key in keys}
+        pass
+
+    def get_chain(self) -> ChainType:
+        """
+        Link Atoms into a processing chain
+
+        Returns
+        -------
+        chain : ProcessingChain
+            The ProcessingChain object that contains
+            additional execution methods.
+
+        """
+        atoms = [atom for _, atom in self.data.items()]
+        chain = ProcessingChain(atoms)
+        return chain
+
+    def execute(
```

Comment:
Yes, it would make sense to have a `__call__` method. With the introduction of the `DataCollections`, the logic of `execute` / `process` will become more complex, so having a minimalistic `__call__` method on the side is warranted.